spring-boot-project/spring-boot/src/main/java/org/springframework/boot/BeanDefinitionLoader.java

```diff
@@ -80,9 +91,13 @@
 		Assert.notEmpty(sources, "Sources must not be empty");
 		this.sources = sources;
 		this.annotatedReader = new AnnotatedBeanDefinitionReader(registry);
-		this.xmlReader = new XmlBeanDefinitionReader(registry);
-		if (isGroovyPresent()) {
-			this.groovyReader = new GroovyBeanDefinitionReader(registry);
+		if (IS_XML_ENABLED) {
+			this.xmlReader = new XmlBeanDefinitionReader(registry);
+			this.groovyReader = isGroovyPresent() ? new GroovyBeanDefinitionReader(registry) : null;
```

Comment:
This seems too broad to me. I don't think that a `spring.xml.ignore` property should have an effect on whether or not `.groovy` bean definitions can be loaded.

Reply:
Indeed, I thought initially that the XML reader was a hard requirement for the Groovy one, but after a deeper look it seems to not be the case, I will update this PR accordingly.